mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -2172,25 +2155,38 @@ pub async fn run_maker_swap(swap: RunMakerSwapInput, ctx: MmArc) {
                     },
                     None => {
                         if let Err(e) = mark_swap_as_finished(ctx.clone(), running_swap.uuid).await {
-                            error!("!mark_swap_finished({}): {}", uuid, e);
+                            error!("!mark_swap_finished({}): {}", uuid_str, e);
                         }
 
                         if to_broadcast {
-                            if let Err(e) = broadcast_my_swap_status(&ctx, uuid).await {
-                                error!("!broadcast_my_swap_status({}): {}", uuid, e);
+                            if let Err(e) = broadcast_my_swap_status(&ctx, running_swap.uuid).await {
+                                error!("!broadcast_my_swap_status({}): {}", uuid_str, e);
                             }
                         }
                         break;
                     },
                 }
             }
         }
-        .fuse(),
-    );
-    select! {
-        _swap = swap_fut => (), // swap finished normally
-        _touch = touch_loop => unreachable!("Touch loop can not stop!"),
-    };
+        .fuse()
+    });
+
+    let (abortable, handle) = abortable(async move {
+        select! {
+         _swap = swap_fut => (), // swap finished normally
+         _touch = touch_loop => unreachable!("Touch loop can not stop!"),
+        }
+    });
+    let uuid = running_swap.uuid;
+    swap_ctx
+        .running_swaps
+        .lock()
+        .unwrap()
+        .insert(uuid, (running_swap, handle.into()));
+    // Wait until the swap has finished (or interrupted, i.e. aborted/panic).
+    if abortable.await.is_err() {
+        info!("Swap uuid={} interrupted!", uuid);
+    }
     // Remove the swap from the running swaps map.
     swap_ctx.running_swaps.lock().unwrap().remove(&uuid);
```

Comment:
there is a side effect introduced in this PR that i don't like:
if the parent of this func call (`run_maker_swap/run_taker_swap`) aborts mid-way, the swap will not abort. one could still abort it ofc from the `running_swaps` map.
but i think it's more intuitive to have the swap aborted when `run_maker_swap` is aborted without the need to abort it via `running_swaps` map.

i think what i want is to be able to run the swap in the current abortable system while still be able to shut the swap using the aborthandle. so there is a 2-way shutdown mechanism or something like that.

cc/ @dimxy

Reply:
Yeah, I think maybe we should create an abortable future (swap_fut+ touch_loop) in run_taker_swap(), add its abort_handle to both the abortable_system and running_swaps.
Then spawn this abortable future and do not await for it.